src/stores/A_store.js

```diff
@@ -71,6 +71,53 @@ class A_store extends ReduceStore {
                 return {...state, stage, path, player: {x: nextLocation.x, y: nextLocation.y}};
             }
 
+            case ActionTypes.GENERATE_FOG: {
+                const {player, end} = state;
+                const {x, y} = player;
+                const revealedGrid = mapGenerator(player, end, new Grid(SIZE, SIZE));
+                const pathGrid = new Grid(SIZE, SIZE);
+
+                let visibleCells = revealedGrid.getVisible(x, y);
+                revealedGrid.obfuscateGrid(x, y, pathGrid, visibleCells);
+                let seenCells;
+                visibleCells = pathGrid.getVisible(x, y);
+                seenCells = pathGrid.getVisible(x, y);
+
+                const start = pathGrid.getCell(x, y);
+                const goal = pathGrid.getCell(end.x, end.y);
+                const path = Astar(start, goal, Cell.heuristic);
+                path.shift();
+
+                return {...state, pathGrid, revealedGrid, path, stage: 'STEP_FOG', visibleCells, seenCells};
+            }
+
+            case ActionTypes.STEP_FOG: {
+                let path = [...state.path];
+                const {revealedGrid, end, seenCells} = state;
+                const nextLocation = path.shift();
+                let stage, pathGrid, visibleCells;
```

Comment:
nit: usually we discourage declaring multiple variables on the same line (outside of destructuring)

Reply:
:rabbit2: